pkg/resources/aws/aws.go

```diff
@@ -1179,47 +1179,34 @@ func ListAutoScalingGroups(cloud fi.Cloud, clusterName string) ([]*resources.Res
 	return resourceTrackers, nil
 }
 
-// FindAutoScalingLaunchTemplateConfigurations finds any launch configurations which reference the security groups
-func FindAutoScalingLaunchTemplateConfigurations(cloud fi.Cloud, securityGroups sets.String) ([]*resources.Resource, error) {
-	var list []*resources.Resource
-
+// FindAutoScalingLaunchTemplates finds any launch configurations which reference the security groups
+func FindAutoScalingLaunchTemplates(cloud fi.Cloud, clusterName string, securityGroups sets.String) ([]*resources.Resource, error) {
 	c, ok := cloud.(awsup.AWSCloud)
 	if !ok {
 		return nil, errors.New("expected a aws cloud provider")
 	}
 	klog.V(2).Infof("Finding all Autoscaling LaunchTemplates associated to security groups")
 
-	resp, err := c.EC2().DescribeLaunchTemplates(&ec2.DescribeLaunchTemplatesInput{MaxResults: fi.Int64(100)})
+	output, err := c.EC2().DescribeLaunchTemplates(&ec2.DescribeLaunchTemplatesInput{
+		Filters: []*ec2.Filter{
+			{
+				Name:   aws.String("tag:KubernetesCluster"),
```

Comment:
Should we be using the legacy tag? Perhaps use the `kubernetes.io/cluster/` one instead?

Reply:
Good point, will change.